launcher/tests/test_notify.py

```diff
@@ -0,0 +1,171 @@
+import datetime
+import os
+import pytest
+import re
+
+from unittest import mock
+from importlib.machinery import SourceFileLoader
+from tempfile import TemporaryDirectory
+
+relpath_notify = "../sdw_notify/Notify.py"
+path_to_notify = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), relpath_notify
+)
+notify = SourceFileLoader("Notify", path_to_notify).load_module()
+
+relpath_updater = "../sdw_updater_gui/Updater.py"
+path_to_updater = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), relpath_updater
+)
+updater = SourceFileLoader("Updater", path_to_updater).load_module()
+
+
+# Regex for warning log if the last-updated timestamp does not exist (updater
+# has never run)
+NO_TIMESTAMP_REGEX = r"Timestamp file '.*' does not exist."
+
+# Regex for warning log if we've updated too long ago, and grace period has elapsed
+UPDATER_WARNING_REGEX = (
+    r"^Last successful update \(.* hours ago\) is above warning threshold "
+)
+r"\(.* hours\). Uptime grace period of .* hours has elapsed (uptime: .* hours)."
```

Comment:
Looks like L30 & L31 should be swapped. As written, `UPDATER_WARNING_REGEX` is resolving to the first half of the pattern. The problem of strings-not-concatenating occurs throughout `test_notify.py`, will comment in-line.

Reply:
Yikes, great flag. `black` helpfully inserted the parentheses, which should have made it obvious to me that the concatenation was wrong. 10003bd31707da6776820006cd3af55f787790c1 should fix these concatenations throughout.

Sometimes Python makes me miss semicolons. ;-)